pczt/src/lib.rs

```diff
@@ -1 +1,150 @@
+//!   - This can be useful e.g. when creating a transaction that has inputs from multiple
+//!     independent Signers; each can receive a PCZT with just the information they need
+//!     to sign, but (e.g.) not the `alpha` values for other Signers.
+//! - Prover
+//!   - Needs all private information for a single spend or output.
+//!   - In practice, the Updater that adds a given spend or output will either act as
+//!     the Prover themselves, or add the necessary data, offload to the Prover, and
+//!     then receive back the PCZT with private data stripped and proof added.
+//! - Signer
+//!   - Needs the spend authorization randomizers to create signatures.
+//!   - Needs sufficient information to verify that the proof is over the correct data.
+//!     without needing to verify the proof itself.
+//!   - A Signer should only need to implement:
+//!     - Pedersen commitments using Jubjub arithmetic (for note and value commitments)
+//!     - BLAKE2b and BLAKE2s (and the various PRFs / CRHs they are used in)
+//!     - Nullifier check (using Jubjub arithmetic)
+//!     - KDF plus note decryption (AEAD_CHACHA20_POLY1305)
+//!     - SignatureHash algorithm
+//!     - Signatures (RedJubjub)
+//!     - A source of randomness.
+//! - Combiner
+//!   - Combines several PCZTs that represent the same transaction into a single PCZT.
+//!   - Because we aren't storing the partial transaction in network format, we need to
+//!     carefully define equality for PCZTs.
+//!     - If we say "pczt.global must be identical" then:
+//!       - If we add spends or outputs in series, we should always update bsk when adding
+//!         spends or outputs, even if rcv is present.
+//!       - If we add spends or outputs in parallel and then combine, we must _never_ update
+//!         bsk, and then update it when we prepare for signature creation.
+//!       We can't control which happens, ergo we need an IO Finalizer step.
+//!     - Once every spend and output has its zkproof field set, PCZT equality MUST include
+//!       the SpendDescription and OutputDescription contents being identical.
+//!       - In practice enforced by creating a TransactionData / CMutableTransaction from
+//!         the PCZT, with spendAuthSigs and bindingSig empty, and then enforcing equality.
+//!       - This is equivalent to BIP 147's equality definition (the partial transactions
+//!         must be identical).
+//! - Spend Finalizer
+//!   - Currently unnecessary, but when shielded multisig is implemented, this would be the
+//!     entity that combines the separate signatures into a multisignature.
+//! - Transaction Extractor
+//!   - Creates bindingSig and extracts the final transaction.
 
+pub mod roles;
+
+mod common;
+mod orchard;
+mod sapling;
+mod transparent;
+
+const V5_TX_VERSION: u32 = 5;
+const V5_VERSION_GROUP_ID: u32 = 0x26A7270A;
+
+/// A partially-created Zcash transaction.
+#[derive(Clone)]
+pub struct Pczt {
```

Comment:
It's very verbose, and the documentation is helpful for explaining what this is to someone that's not already familiar, but could this be renamed to `TransactionParts` or something that's more immediately understandable without context?

Reply:
- `UnsignedTransaction` is incorrect; the PCZT format can contain data that on its own cannot fully produce an unsigned transaction.
- `PartialTransaction` might be more reasonable (as a shortening of `PartiallyCreatedZcashTransaction`), but it doesn't clarify in what way the transaction is partial.
- `PartiallyCreatedTransaction` would be another option (just expanding `Pczt` and dropping the superfluous Zcash for which we already have the context from the `pczt` crate name).

In any case, this choice is immaterial to the discussion at hand, which is the PCZT format semantics.